### ISSUE TITLE
How to load resources on-demand and sync it with events?

### ISSUE THREAD

**Issue by coirpower** [vkurko/calendar#376]
`<script>
    document.addEventListener('DOMContentLoaded', function() {

        // Embed resources directly using PHP variable
        var resources = @json($staffs);

        var calendarEl = document.getElementById('calendar');

        var calendar = new EventCalendar(calendarEl, {
            view: 'resourceTimelineWeek',
            firstDay: 1,
            slotDuration: {days: 1},
            headerToolbar: {
                start: 'prev,next today',
                center: 'title',
                end: 'resourceTimelineWeek'
            },
            eventSources: [
                {
                    url: '/attendance/resourcesAndAttendance',
                    method: 'GET',
                    failure: function() {
                        alert('There was an error while fetching events!');
                    }
                }
            ],
            views: {
                resourceTimelineWeek: {
                    pointer: true, 
                    resources: resources
                }
            },
            dayMaxEvents: true,
            nowIndicator: true,
            selectable: false            
        });
    });
</script>`

I use the above code but it looks like the calendar loads resources at initialization and does not automatically refresh them when I fetch events. 

Basically, I have server side code that will return only the required resources and their events, and would like the calendar to fetch and load these dataset at runtime. Any suggestions?

**Comment by vkurko** [comment]
It seems that something like [refetchResources()](https://fullcalendar.io/docs/refetchResources) is needed. I will try to implement it in future versions.

**Comment by busti** [comment]
I wound up using a workaround:  

```
        calendar.setOption('eventSources',
            [
                {
                    events: function (fetchInfo, successCallback, failureCallback) {
                        req = fetch(`/schedule/events?start=${fetchInfo.startStr}&end=${fetchInfo.endStr}`)
                        json = req.then(res => res.json())
                        json.then(j => {
                            window.history.pushState(null, '', `/schedule?start=${fetchInfo.startStr}&end=${fetchInfo.endStr}`);
                            calendar.setOption('resources', j.resources);
                            successCallback(j.events);
                        })

                        req.catch(e => failureCallback(e))
                    }
                }
            ]
        )
```

It's not perfect tho. Resources can shuffle around and collapsed resources open back up.

**Comment by vkurko** [comment]
This should be implemented in v5.3.0. Please see [resources](https://github.com/vkurko/calendar?tab=readme-ov-file#resources) and [refetchResources()](https://github.com/vkurko/calendar?tab=readme-ov-file#refetchresources).
